src/common/post.ts

```diff
@@ -329,3 +344,125 @@ export const insertCodeSnippetsFromUrl = async ({
     throw err;
   }
 };
+
+type ModeratedPostCdc = ChangeObject<SourcePostModeration>;
+
+export const updateModeratedPost = async (
+  con: ConnectionManager,
+  moderated: ModeratedPostCdc,
+): Promise<ModeratedPostCdc | null> => {
+  if (!moderated?.postId) {
+    logger.error('unable to update moderated post', { moderated });
+    return null;
+  }
+
+  const postParam = { id: moderated.postId };
+  const repo = con.getRepository(Post);
+
+  await repo.findOneOrFail({ where: { id: moderated.postId } });
+
+  const updatedPost: Partial<FreeformPost> = {
+    title: moderated.title,
+    titleHtml: moderated.titleHtml,
+  };
+
+  if (moderated.content) {
+    updatedPost.content = moderated.content;
+    updatedPost.contentHtml = moderated.contentHtml!;
+  }
+
+  if (moderated.image) {
+    updatedPost.image = moderated.image;
+  }
+
+  await repo.update(postParam, updatedPost);
+
+  return moderated;
+};
+
+export const getExistingPost = async (
+  manager: ConnectionManager,
+  cleanUrl: string,
+): Promise<Post | null> =>
+  await manager
+    .createQueryBuilder(Post, 'post')
+    .select(['post.id', 'post.deleted', 'post.visible'])
+    .where('post.url = :url OR post.canonicalUrl = :url', {
+      url: cleanUrl,
+    })
+    .getOne();
+
+export const processApprovedModeratedPost = async (
+  con: ConnectionManager,
+  moderated: ModeratedPostCdc,
+): Promise<ModeratedPostCdc | null> => {
+  if (!moderated) {
+    throw new Error('Moderated post is missing');
+  }
+
+  if (moderated.postId) {
+    return updateModeratedPost(con, moderated);
+  }
+
+  const {
+    title,
+    content,
+    contentHtml,
+    image,
+    createdById,
+    sourceId,
+    titleHtml,
+    externalLink,
+    sharedPostId,
+  } = moderated;
+
+  if (moderated.type === PostType.Freeform) {
+    const id = await generateShortId();
+    const params = {
+      id,
+      title,
+      titleHtml,
+      content,
+      contentHtml,
+      image,
+      sourceId,
+      authorId: createdById,
+    };
+    const post = await createFreeformPost({ con, args: params as CreatePost });
+    return { ...moderated, postId: post.id };
+  }
+
+  if (sharedPostId) {
+    const post = await createSharePost({
+      con,
+      args: {
+        postId: sharedPostId,
+        commentary: title,
+        authorId: createdById,
+        sourceId,
+      },
+    });
+
+    return { ...moderated, postId: post.id };
+  }
+
+  if (externalLink) {
+    const cleanUrl = standardizeURL(externalLink);
+    const post = await createExternalLink({
+      con,
+      args: {
+        title,
+        image,
+        url: cleanUrl,
+        sourceId,
+        authorId: createdById,
+        commentary: content,
+      },
+    });
+    return { ...moderated, postId: post.id };
+  }
+
+  logger.error(`unable to process moderated post: ${moderated.id}`);
```

Comment:
Let's use object style syntax as we use in other workers, it will give us better visibility

```suggestion
  logger.error({ moderated }, 'unable to process moderated post');
```

Reply:
Cool. Didn't wanna flood or logs with might be unnecessary props which can just be pulled.

Reply:
This should basically never happen, so if when deploy it starts to happen we need to adjust the worker so it handles the cases which end up on this

Reply:
Makes sense, yeah. I'll update quickly.